webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -658,8 +661,14 @@ class PolicyWriterServiceImpl(
       // Folders is a map of machine.uuid -> (base_machine_folder, backup_machine_folder, machine)
       for (folder @ NodePromisesPaths(_, baseFolder, newFolder, backupFolder) <- sortedFolder) {
         // backup old promises
-        logger.trace("Backuping old promises from %s to %s ".format(baseFolder, backupFolder))
-        backupNodeFolder(baseFolder, backupFolder)
+        if (backupPoliciesAtEndOfGeneration) {
+          logger.trace("Backuping old policies from %s to %s ".format(baseFolder, backupFolder))
+          backupNodeFolder(baseFolder, backupFolder)
+        } else {
+          logger.trace(s"Deleting old policies in ${baseFolder}")
+          deleteNodeFolder(baseFolder)
```

Comment:
shouldn't we also delete backupFolder if it exits from a previous generation with a different backup strategy?
Else, people may become confuse when/if they find back the olds generated files at some point.

Reply:
yes, that's a good idea